spec/my_enumerable_module_spec.rb

```diff
@@ -0,0 +1,43 @@
+require ('./my_enumerable_module.rb')
+
+describe Enumerable do 
+
+  describe '#my_each' do 
+
+    it 'iterates through each element of the object passed to it' do
+      expect(['hi', 'hello', 1, 2].my_each {|idx| puts idx }).to eq(['hi', 'hello', 1, 2])
+    end
+
+    it 'does nothing if not block is given' do
+      expect([].my_each).to eq([])
+    end
+
+  end
+
+  describe '#my_each_with_index' do
+
+    it 'returns each item and index in the object' do
+      expect(['hi', 'hello'].my_each_with_index {|val, idx| puts "index: #{idx} for val: #{val}" }).to eq(['hi', 'hello'])
+    end
+
+  end 
+
+  describe '#my_select' do 
+
+    it 'selects items out of the object that appear in the search regex' do
+      expect(['a', 'b', 'c', 'd', 'e'].my_select {|i| i =~ /[aeiou]/}).to eq(['a', 'e'])
+    end
+
```

Comment:
Extra blank line.

Reply:
is there a way to easily spot blank lines? i have a lot of them and highlighting all the text in my editor doesn't seem to be a clear way for me to spot them.. hmm..

Reply:
Well ... one way is to search for a regex. A typical regex for an editor for an empty line or a line with only blanks might be something like ...
```
/^\s*$/
```

So `^` start of line, `\s` is a space, `*` is 0 or more, and `$` is end of line. You should get all of them that way I think.

Reply:
Oh I didn't know you can search for a regular expression, thank you! 👍🏻